### PR TITLE
Custom Grant type

### PR DESCRIPTION
**Pull request by Nakira**
Just did a phpcs forked from https://github.com/zfcampus/zf-oauth2/pull/116

### CODE REVIEW

File: src/Factory/OAuth2ServerInstanceFactory.php [hunk 1]
```diff
@@ -141,6 +141,15 @@ public function __invoke()
             $server->addGrantType(new RefreshToken($server->getStorage('refresh_token'), $refreshOptions));
         }
 
+        // Add custom grant type from the service locator
+        if (isset($availableGrantTypes['custom_grant_types']) && is_array($availableGrantTypes['custom_grant_types'])) {
+            foreach ($availableGrantTypes['custom_grant_types'] as $grantKey => $grantType) {
+                if ($services->has($grantType)) {
+                    $server->addGrantType($services->get($grantType, $grantKey));
```
Comment (hkulekci) [inline src/Factory/OAuth2ServerInstanceFactory.php]: should this be like `$this->service->has($grantType)` and `$this->services->get($grantType, $grantKey)`?
Am I missing something?

**Comment by iam-merlin** [comment]
Yes, you right. My code was just a proposal.

For one of our project, I move my change to mvc-auth that you can [consult here](https://github.com/eoko/zf-mvc-auth) and especially in [this commit](https://github.com/eoko/zf-mvc-auth/commit/1560c775e26eb07e024c07649eb8ff57be5d0b7a).

If you have any questions, don't hesitate to ask. This code is running for a client, it works fine but it's just a POC.

**Comment by iam-merlin** [comment]
By the way, I close this issue in favor of : https://github.com/zfcampus/zf-mvc-auth/pull/105

**Comment by Nakira** [comment]
Ok following @iam-merlin on this. I close this PR.

**Pull request closed without merging**